Dockerfile

```diff
@@ -0,0 +1,60 @@
+# ---------- 1st stage ----------
+# The first stage adds the necessary libraries to build native add-ons (eg. bcrypt) and then installs
+# the server dependencies
+FROM node:12-alpine as builder
+
+# By default, `alpine` images don't have necessary tools to install native add-ons, so we use the
+# multistage build to install the necessary tools, and build the dependencies which will then be
+# copied over to the next stage (this stage will be discarded, along with the installed tools, to
+# make the image lighter).
+RUN apk add --no-cache python make g++ git
+
+# Set the working directory to where we want to install the dependencies; this directory doesn't
+# really matter as it will not be present in the final image, but keeping the path short makes
+# copying from it easier in the next stage.
+WORKDIR /shieldbattery
+
+# Copy the files necessary to install dependencies
+COPY server/package.json server/
+COPY server/yarn.lock server/
+COPY package.json ./
+COPY yarn.lock ./
+
+# Install only the server and main dependencies (`app` dependencies should be built into the Electron app)
+# TODO(2Pac): Figure out why some `devDependencies` are needed for the server to run in production
```

Comment:
Seems to me like these are just incorrectly tagged as dev dependencies, they're actually used in production as well (because we don't precompile with babel, although I guess we could? Might make debugging server errors a bit harder though).

If we were willing to move to `.mjs` files and required that the production server ran webpack separately (e.g. it expected that the compiled client JS was already there rather than compiling it on startup) we could probably do away with babel as a production dep for the server. Not sure how big of a gain it really is though, the `@babel` folder in my node_modules is like 13MB. (It *is* potentially nice from a runtime speed and reproducible build perspective, just not sure it's worth the effort atm)